drivers/at/at.c

```diff
@@ -492,63 +512,53 @@ static size_t at_readline_stop_at_str(at_dev_t *dev, char *resp_buf, size_t len,
                                       bool keep_eol, char const *substr,
                                       uint32_t timeout)
 {
-    const char eol[] = AT_RECV_EOL_1 AT_RECV_EOL_2;
-    assert(sizeof(eol) > 1);
-
     ssize_t res = 0;
-    char *resp_pos = resp_buf;
-
-#if IS_USED(MODULE_AT_URC_ISR)
-    dev->awaiting_response = true;
-#endif
-
-    memset(resp_buf, 0, len);
     size_t substr_len = 0;
+
     if (substr) {
         substr_len = strlen(substr);
         if (substr_len == 0) {
             return -EINVAL;
         }
     }
+
+    char *resp_pos = resp_buf;
     char const *substr_p = resp_buf;
+    memset(resp_buf, 0, len);
+
     while (len > 1) {
-        int read_res;
-        if ((read_res = isrpipe_read_timeout(&dev->isrpipe, (uint8_t *)resp_pos,
-                                             1, timeout)) == 1) {
-            if (AT_PRINT_INCOMING) {
-                print(resp_pos, read_res);
-            }
-            if (sizeof(eol) > 2 && *resp_pos == eol[0]) {
-                if (!keep_eol) {
-                    continue;
-                }
-            }
-            if (*resp_pos == eol[sizeof(eol) - 2]) {
-                *resp_pos = '\0';
-                break;
-            }
+        res = isrpipe_read_timeout(&dev->isrpipe, (uint8_t *)resp_pos, 1, timeout);
```

Comment:
reading byte by byte might slow this down (ztimer_set happening for every byte)

Reply:
This was originally so but I agree. The solution would be to insert yet another buffer in-between. I would address in another PR, as it doesn't bring any functional changes and this one is already large enough.